### ISSUE TITLE
implicit statement is unsupported

### ISSUE THREAD

**Issue by kargl** [lfortran/lfortran#1675]
```
   subroutine print_time
      implicit real (a-h,p-z)
   end subroutine print_time

 ~/lfortran/work/bin/lfortran -c a.f90
semantic error: Implicit typing is not allowed, enable it by using --implicit-typing 
 --> a.f90:2:7
  |
2 |       implicit real (a-h,p-z)
  |       ^^^^^^^^^^^^^^^^^^^^^^^^ 

Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
```

IMPLICIT statement is covered in Fortran 2018, 8.7 IMPLICIT statement.

**Comment by certik** [comment]
You currently have to enable it with `--implicit-typing` like this:
```console
$ lfortran -c a.f90 --implicit-typing
```
Then it works. If you use implicit interfaces, also add `--implicit-interface`.

I am going to keep this issue open, as there might not be any harm allowing this by default, if the "implicit" statement is used.

This is also tied to issue https://github.com/lfortran/lfortran/issues/1677.

**Comment by kargl** [comment]
IMHO, standard conforming code should not require an option to allow the construct to be compiled.  If one has
```
subroutine foo
    x = 42
    print *, x
end
```
This should compile without an option.  lfortran can emit a loud warning to complain about an implicit entity, but it should compile.  An option can then be provided to silences the warning.

**Comment by certik** [comment]
That's one way to do it. But you would have to then provide the option to silence the warning for older code, which is almost as annoying as the current approach. 

Another way is to provide a `f23` binary, which would have all the option on by default and not even warn about it. 

We'll figure this out before a 1.0 release, once we can compile most codes.

**Comment by kargl** [comment]
Yes, having a `-std=f23` or `-std=f18` with all the options on would be a good compromise.  I've suggested switching gfortran to `-std=f2018` (and/or `f2023` when available), but that did not go over too well.

What I do find interesting is that one needs to use `--implicit-typing` to compile standard conforming code while `real*4 x` is silently swallowed by lfortran, and AFAICT there is no option to turn on a warning or error.

**Comment by certik** [comment]
I'll try to do the `-std=f23` options soon, and we'll see it's good enough, or if we need to iterate more on this.

Yes, we need to emit warnings for `real*4 x`, I created an issue https://github.com/lfortran/lfortran/issues/1680 for that.
